internal/rbd/nodeserver.go

```diff
@@ -149,6 +149,65 @@ func healerStageTransaction(ctx context.Context, cr *util.Credentials, volOps *r
 	return nil
 }
 
+// populateVolOptions update the fields in rbdVolume struct based on the request it received.
+func (ns *NodeServer) populateVolOptions(
+	ctx context.Context,
+	req *csi.NodeStageVolumeRequest,
+	disableInUseChecks bool,
+	isStaticVol bool,
+	volID string, cr *util.Credentials) (*rbdVolume, error) {
+	var err error
+	volOptions, err := genVolFromVolumeOptions(ctx, req.GetVolumeContext(), req.GetSecrets(), disableInUseChecks)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	volOptions.ThickProvision = isThickProvisionRequest(req.GetVolumeContext())
+
+	// get rbd image name from the volume journal
+	// for static volumes, the image name is actually the volume ID itself
+	if isStaticVol {
+		volOptions.RbdImageName = volID
+	} else {
+		var vi util.CSIIdentifier
+		var imageAttributes *journal.ImageAttributes
+		err = vi.DecomposeCSIID(volID)
+		if err != nil {
+			err = fmt.Errorf("error decoding volume ID (%s): %w", volID, err)
+
+			return nil, status.Error(codes.Internal, err.Error())
+		}
+
+		j, connErr := volJournal.Connect(volOptions.Monitors, volOptions.RadosNamespace, cr)
+		if connErr != nil {
+			log.ErrorLog(ctx, "failed to establish cluster connection: %v", connErr)
```

Comment:
consider updating `connErr` just like is done for other error cases

Reply:
done.